front/src/app/error.tsx

```diff
@@ -0,0 +1,23 @@
+"use client";
+
+import { useEffect } from "react";
+import Header from "./Header";
+
+export default function Error({
+    error,
+    reset,
+}: {
+    error: Error;
+    reset: () => void;
+}) {
+    useEffect(() => {
+        console.error(error);
+    }, [error]);
+
+    return (
+        <Header>
+            <h2>Something went wrong!</h2>
+            <button onClick={() => reset()}>Try again</button>
+        </Header>
```

Comment:
Encapsulating the body in the header seems wrong... It might be better to rename it PageLayout or something more generic, don't you think? Otherwise we can have the Header as a component that we add at the top of the page, and keep the `<main>` everywhere...